inlanecruising_plugin/include/inlanecruising_plugin/smoothing/SplineI.h

```diff
@@ -43,13 +43,31 @@ class SplineI
   virtual void setPoints(std::vector<lanelet::BasicPoint2d> points) = 0;
   
     /**
-   * \brief Get the BasicPoint2d coordinate along the curve at t-th step. 
+  *  \brief Get the BasicPoint2d coordinate along the curve at t-th step. 
    * 
    * \param t The t-th step to solve the spline at, where t is from 0 (beginning of curve) to 1 (end of curve)
    * 
    * \return lanelet::BasicPoint2d with x, y that matches the t-th step along the curve
    */ 
   virtual lanelet::BasicPoint2d operator()(double t) const = 0;
+
+  /**
+  *  \brief Get the BasicPoint2d representing the first_deriv along the curve at t-th step. 
+   * 
+   * \param t The t-th step to solve the spline at, where t is from 0 (beginning of curve) to 1 (end of curve)
+   * 
+   * \return lanelet::BasicPoint2d with x, y that matches the first_deriv at t-th step along the curve
```

Comment:
Comment should clarify if these are partial derivatives. Are they?

Reply:
I don't think these are partial derivatives. I commented to clarify